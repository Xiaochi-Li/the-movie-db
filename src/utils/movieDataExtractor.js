import { pick } from "lodash";

export function pickMovieProperties(OriginalMovie) {
  const MOVIE_PROPERTIES = [
    "id",
    "vote_average",
    "title",
    "poster_path",
    "backdrop_path",
    "overview"
  ];

  const movieProperties = pick(OriginalMovie, MOVIE_PROPERTIES);

  const {
    id,
    vote_average,
    title,
    poster_path,
    backdrop_path,
    overview
  } = movieProperties;

  const movie = {
    voteAverage: vote_average,
    posterPath: poster_path,
    backdropPath: backdrop_path,
    id,
    title,
    overview
  };

  return movie;
}

export function getTrimmedMovieList(OriginalMovieList) {
  const trimmedMoviewList = OriginalMovieList.map(movie =>
    pickMovieProperties(movie)
  );
  return trimmedMoviewList;
}
