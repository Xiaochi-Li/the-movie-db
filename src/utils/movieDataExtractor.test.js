import { getTrimmedMovieList } from "./movieDataExtractor";
import originalForms from "./__fixtures__/testPopularMoviesData";

it("should return a simplified JSON form", () => {
  expect.addSnapshotSerializer({
    test: val => true, // is this needed?
    print: val => `${JSON.stringify(val, null, 2)}`
  });

  expect(getTrimmedMovieList(originalForms)).toMatchSnapshot();
});
