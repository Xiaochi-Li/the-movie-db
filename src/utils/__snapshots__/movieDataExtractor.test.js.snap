// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a simplified JSON form 1`] = `
[
  {
    "voteAverage": 7.7,
    "posterPath": "https://image.tmdb.org/t/p/w154//xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
    "releaseDate": "January 2019",
    "id": 166428,
    "title": "How to Train Your Dragon: The Hidden World",
    "overview": "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind."
  },
  {
    "voteAverage": 7.2,
    "posterPath": "https://image.tmdb.org/t/p/w154//AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
    "releaseDate": "March 2019",
    "id": 299537,
    "title": "Captain Marvel",
    "overview": "The story follows Carol Danvers as she becomes one of the universe’s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races. Set in the 1990s, Captain Marvel is an all-new adventure from a previously unseen period in the history of the Marvel Cinematic Universe."
  },
  {
    "voteAverage": 6.5,
    "posterPath": "https://image.tmdb.org/t/p/w154//svIDTNUoajS8dLEo7EosxvyAsgJ.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//svIDTNUoajS8dLEo7EosxvyAsgJ.jpg",
    "releaseDate": "January 2019",
    "id": 450465,
    "title": "Glass",
    "overview": "In a series of escalating encounters, security guard David Dunn uses his supernatural abilities to track Kevin Wendell Crumb, a disturbed man who has twenty-four personalities. Meanwhile, the shadowy presence of Elijah Price emerges as an orchestrator who holds secrets critical to both men."
  },
  {
    "voteAverage": 6.8,
    "posterPath": "https://image.tmdb.org/t/p/w154//iYwKQAO1LwLpInMMdvmYChrRlHN.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//iYwKQAO1LwLpInMMdvmYChrRlHN.jpg",
    "releaseDate": "March 2019",
    "id": 329996,
    "title": "Dumbo",
    "overview": "A young elephant, whose oversized ears enable him to fly, helps save a struggling circus, but when the circus plans a new venture, Dumbo and his friends discover dark secrets beneath its shiny veneer."
  },
  {
    "voteAverage": 7.4,
    "posterPath": "https://image.tmdb.org/t/p/w154//ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
    "releaseDate": "March 2019",
    "id": 458723,
    "title": "Us",
    "overview": "Husband and wife Gabe and Adelaide Wilson take their kids to their beach house expecting to unplug and unwind with friends. But as night descends, their serenity turns to tension and chaos when some shocking visitors arrive uninvited."
  },
  {
    "voteAverage": 6.5,
    "posterPath": "https://image.tmdb.org/t/p/w154//fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
    "releaseDate": "December 2018",
    "id": 424783,
    "title": "Bumblebee",
    "overview": "On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug."
  },
  {
    "voteAverage": 7.4,
    "posterPath": "https://image.tmdb.org/t/p/w154//xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
    "releaseDate": "March 2019",
    "id": 287947,
    "title": "Shazam!",
    "overview": "A boy is given the ability to become an adult superhero in times of need with a single magic word."
  },
  {
    "voteAverage": 0,
    "posterPath": "https://image.tmdb.org/t/p/w154//dHjLaIUHXcMBt7YxK1TKWK1end9.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//dHjLaIUHXcMBt7YxK1TKWK1end9.jpg",
    "releaseDate": "April 2019",
    "id": 299534,
    "title": "Avengers: Endgame",
    "overview": "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store."
  },
  {
    "voteAverage": 8.3,
    "posterPath": "https://image.tmdb.org/t/p/w154//7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
    "releaseDate": "April 2018",
    "id": 299536,
    "title": "Avengers: Infinity War",
    "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain."
  },
  {
    "voteAverage": 6.8,
    "posterPath": "https://image.tmdb.org/t/p/w154//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
    "releaseDate": "December 2018",
    "id": 297802,
    "title": "Aquaman",
    "overview": "Once home to the most advanced civilization on Earth, Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people and then the surface world. Standing in his way is Arthur Curry, Orm's half-human, half-Atlantean brother and true heir to the throne."
  },
  {
    "voteAverage": 7,
    "posterPath": "https://image.tmdb.org/t/p/w154//4bRYg4l12yDuJvAfqvUOPnBrxno.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//4bRYg4l12yDuJvAfqvUOPnBrxno.jpg",
    "releaseDate": "March 2019",
    "id": 500682,
    "title": "The Highwaymen",
    "overview": "In 1934, Frank Hamer and Manny Gault, two former Texas Rangers, are commissioned to put an end to the wave of vicious crimes perpetrated by Bonnie Parker and Clyde Barrow, a notorious duo of infamous robbers and cold-blooded killers who nevertheless are worshiped by the public."
  },
  {
    "voteAverage": 6,
    "posterPath": "https://image.tmdb.org/t/p/w154//7SPhr7Qj39vbnfF9O2qHRYaKHAL.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//7SPhr7Qj39vbnfF9O2qHRYaKHAL.jpg",
    "releaseDate": "April 2019",
    "id": 157433,
    "title": "Pet Sematary",
    "overview": "Louis Creed, his wife Rachel and their two children Gage and Ellie move to a rural home where they are welcomed and enlightened about the eerie 'Pet Sematary' located nearby. After the tragedy of their cat being killed by a truck, Louis resorts to burying it in the mysterious pet cemetery, which is definitely not as it seems, as it proves to the Creeds that sometimes dead is better."
  },
  {
    "voteAverage": 6.5,
    "posterPath": "https://image.tmdb.org/t/p/w154//oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
    "releaseDate": "December 2018",
    "id": 504172,
    "title": "The Mule",
    "overview": "Earl Stone, a man in his 80s who is broke, alone, and facing foreclosure of his business when he is offered a job that simply requires him to drive. Easy enough, but, unbeknownst to Earl, he’s just signed on as a drug courier for a Mexican cartel. He does so well that his cargo increases exponentially, and Earl hit the radar of hard-charging DEA agent Colin Bates."
  },
  {
    "voteAverage": 4.8,
    "posterPath": "https://image.tmdb.org/t/p/w154//4tS0iyKQBDFqVpVcH21MSJwXZdq.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//4tS0iyKQBDFqVpVcH21MSJwXZdq.jpg",
    "releaseDate": "January 2019",
    "id": 390634,
    "title": "Fate/stay night: Heaven’s Feel II. lost butterfly",
    "overview": "Theatrical-release adaptation of the visual novel \\"Fate/stay night\\", following the third and final route. (Part 2 of a trilogy.)"
  },
  {
    "voteAverage": 6.4,
    "posterPath": "https://image.tmdb.org/t/p/w154//8yZAx7tlKRZIg7pJfaPhl00yHIQ.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//8yZAx7tlKRZIg7pJfaPhl00yHIQ.jpg",
    "releaseDate": "January 2019",
    "id": 522681,
    "title": "Escape Room",
    "overview": "Six strangers find themselves in circumstances beyond their control, and must use their wits to survive."
  },
  {
    "voteAverage": 5,
    "posterPath": "https://image.tmdb.org/t/p/w154//avoKZfgBzyBGJsrAr2WQOwZK978.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//avoKZfgBzyBGJsrAr2WQOwZK978.jpg",
    "releaseDate": "March 2019",
    "id": 500904,
    "title": "A Vigilante",
    "overview": "A vigilante helps victims escape their domestic abusers."
  },
  {
    "voteAverage": 6.9,
    "posterPath": "https://image.tmdb.org/t/p/w154//fMMrl8fD9gRCFJvsx0SuFwkEOop.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//fMMrl8fD9gRCFJvsx0SuFwkEOop.jpg",
    "releaseDate": "November 2018",
    "id": 338952,
    "title": "Fantastic Beasts: The Crimes of Grindelwald",
    "overview": "Gellert Grindelwald has escaped imprisonment and has begun gathering followers to his cause—elevating wizards above all non-magical beings. The only one capable of putting a stop to him is the wizard he once called his closest friend, Albus Dumbledore. However, Dumbledore will need to seek help from the wizard who had thwarted Grindelwald once before, his former student Newt Scamander, who agrees to help, unaware of the dangers that lie ahead. Lines are drawn as love and loyalty are tested, even among the truest friends and family, in an increasingly divided wizarding world."
  },
  {
    "voteAverage": 6.6,
    "posterPath": "https://image.tmdb.org/t/p/w154//xRWht48C2V8XNfzvPehyClOvDni.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//xRWht48C2V8XNfzvPehyClOvDni.jpg",
    "releaseDate": "January 2019",
    "id": 399579,
    "title": "Alita: Battle Angel",
    "overview": "When Alita awakens with no memory of who she is in a future world she does not recognize, she is taken in by Ido, a compassionate doctor who realizes that somewhere in this abandoned cyborg shell is the heart and soul of a young woman with an extraordinary past."
  },
  {
    "voteAverage": 7.4,
    "posterPath": "https://image.tmdb.org/t/p/w154//hQvf3RHgmp4XXXl2y6zhMe4G4kg.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//hQvf3RHgmp4XXXl2y6zhMe4G4kg.jpg",
    "releaseDate": "March 2019",
    "id": 576071,
    "title": "Unplanned",
    "overview": "As one of the youngest Planned Parenthood clinic directors in the nation, Abby Johnson was involved in upwards of 22,000 abortions and counseled countless women on their reproductive choices. Her passion surrounding a woman's right to choose led her to become a spokesperson for Planned Parenthood, fighting to enact legislation for the cause she so deeply believed in. Until the day she saw something that changed everything."
  },
  {
    "voteAverage": 5,
    "posterPath": "https://image.tmdb.org/t/p/w154//2WfjB6FUDTIBVI2y02UGbnHR82s.jpg",
    "largePoster": "https://image.tmdb.org/t/p/w500//2WfjB6FUDTIBVI2y02UGbnHR82s.jpg",
    "releaseDate": "December 2018",
    "id": 450001,
    "title": "Master Z: Ip Man Legacy",
    "overview": "After being defeated by Ip Man, Cheung Tin Chi is attempting to keep a low profile. While going about his business, he gets into a fight with a foreigner by the name of Davidson, who is a big boss behind the bar district. Tin Chi fights hard with Wing Chun and earns respect."
  }
]
`;
